lib/services/files-api-service.js

```diff
@@ -0,0 +1,107 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+var fs = require('fs');
+var path = require('path');
+var express = require('express');
+var app = express();
+var walking = require('walk');
+
+module.exports = filesApiServiceFactory;
+
+di.annotate(filesApiServiceFactory, new di.Provide('Http.Services.Api.Files'));
+di.annotate(filesApiServiceFactory,
+    new di.Inject(
+      'Promise',
+      'Services.Waterline'
+    )
+);
+
+
+function filesApiServiceFactory(
+  Promise,
+  waterline
+) {
+  var walker;
+  var walk = Promise.promisifyAll(walking);
+  var onhttpPath = __dirname.split('/');
+  onhttpPath.splice(0,2);
+  onhttpPath = onhttpPath.join("/");
+  Promise.promisifyAll(fs);
+
+  function FilesApiService() {
+  }
+
+  /**
+   * Get list of static files
+   * @returns {array}
+   */
+
+  FilesApiService.prototype.getAllStaticFiles = function() {
+
+    app.use(express.static(onhttpPath));
+    var staticPath = "static/http";
+    var skuPath = "skupack.d";
+    var directories = [staticPath,
+                       skuPath,
+                       "/opt/monorail/static/http"];
+
+    var skuInfo = {};
+    fs.readdirAsync(skuPath).then(function(list) {
+      list.forEach(function(item){
+        if (fs.lstatSync(path.join(skuPath, item)).isDirectory()){
```

Comment:
You can probably use `Promise.filter` here (http://bluebirdjs.com/docs/api/promise.filter.html):

```
fs.readdirAsync(skuPath).then(function(list) {
    return Promise.filter(list, function(item) {
        return (fs.lstatAsync(path.join(skuPath, item))
        .then(function(stat) {
            return stat.isDirectory();
        });
    })
}).then(function(item) {
    return waterline.skus.findOne({id : item});
}).then(function(sku) { 
        skuInfo[item] = sku.name;
});
```